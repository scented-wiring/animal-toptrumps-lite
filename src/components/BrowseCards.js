import { useState } from "react";
import { Link } from "react-router-dom";
import { cards } from "../cards/index";
import "../styles/BrowseCards.css";
import Card from "./Card";

const BrowseCards = () => {
  const initialState = {
    alert: {
      message: "",
      alertType: "",
    },
    card: {
      message: "Choose a card to view its stats.",
      defaultText: true,
    },
  };

  const [card, setCard] = useState(initialState.card);

  return (
    <div id="BrowseCards">
      <h2>Browse Cards</h2>
      <div id="display">
        <select
          name="selectCards-drop"
          id="selectCards-drop"
          onChange={(e) =>
            setCard(
              cards[document.getElementById("selectCards-drop").selectedIndex]
            )
          }
        >
          {cards.map((card) => (
            <option key={cards.indexOf(card)}>{card.name}</option>
          ))}
        </select>
        <select
          name="selectCards"
          id="selectCards"
          size="15"
          onChange={(e) =>
            setCard(cards[document.getElementById("selectCards").selectedIndex])
          }
        >
          {cards.map((card) => (
            <option key={cards.indexOf(card)}>{card.name}</option>
          ))}
        </select>
        <Card {...card} />
      </div>
      <Link to="/">
        <div className="back">&larr; BACK</div>
      </Link>
    </div>
  );
};

export default BrowseCards;
