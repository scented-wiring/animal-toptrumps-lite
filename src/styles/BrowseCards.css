#EditDeck {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  height: 100%;
}

#display {
  display: flex;
}

#selectCards {
  margin-right: 30px;
  font-size: 23px;
  font-family: "Montserrat", sans-serif;
  height: 460px;
  width: 230px;
  border: 2px black solid;
  box-shadow: 5px 5px #004742;
}

#selectCards:focus {
  outline: 0;
}

#selectCards-drop {
  display: none;
}

@media only screen and (max-width: 600px) {
  #display {
    flex-direction: column;
    align-items: center;
  }

  #selectCards {
    display: none;
  }

  #selectCards-drop {
    width: 200px;
    display: inline;
    margin-bottom: 15px;
    font-family: "Montserrat", sans-serif;
    border: 2px black solid;
    box-shadow: 2px 2px #004742;
  }

  #selectCards-drop:focus {
    outline: none;
  }

  #EditDeck .Card,
  #EditDeck .Card-border {
    height: 400px;
    width: 260px;
  }

  #EditDeck .name {
    font-size: 40px;
    margin-top: 10px;
  }

  #EditDeck .aka {
    font-size: 20px;
  }

  #EditDeck .alignment {
    font-size: 15px;
    margin-bottom: 10px;
  }

  #EditDeck #photo {
    margin: 5px auto 10px auto;
    height: 165px;
    width: 220px;
  }

  #EditDeck .stat-name1,
  #EditDeck .stat-name2 {
    padding-left: 12px;
    height: 25px;
    font-size: 18px;
  }

  #EditDeck .stat1,
  #EditDeck .stat2 {
    height: 25px;
    width: 45px;
  }

  #EditDeck .footer {
    font-size: 8px;
    margin-top: 4px;
  }

  #EditDeck .loading {
    padding: 25px 0;
    height: 30px;
    width: 120px;
    margin: 10px auto 10px auto;
    font-size: 15px;
  }

  #EditDeck .default-message {
    margin-top: 50px;
    font-size: 35px;
  }
}

@media only screen and (max-width: 400px) {
  #EditDeck .Card,
  #EditDeck .Card-border {
    height: 268px;
    width: 175px;
  }

  #EditDeck .name {
    font-size: 25px;
    margin-top: 5px;
  }

  #EditDeck .aka {
    font-size: 14px;
  }

  #EditDeck .alignment {
    font-size: 11px;
    margin-bottom: 0px;
  }

  #EditDeck #photo {
    margin: 5px auto 5px auto;
    height: 96px;
    width: 128px;
  }

  #EditDeck .stat-name1,
  #EditDeck .stat-name2 {
    padding-left: 8px;
    height: 20px;
    font-size: 15px;
  }

  #EditDeck .stat1,
  #EditDeck .stat2 {
    height: 20px;
    width: 30px;
  }

  #EditDeck .footer {
    font-size: 7px;
    margin-top: 4px;
  }

  #EditDeck .loading {
    padding: 25px 0;
    height: 30px;
    width: 120px;
    margin: 10px auto 10px auto;
    font-size: 15px;
  }

  #EditDeck .default-message {
    margin-top: 50px;
    font-size: 35px;
  }
}
