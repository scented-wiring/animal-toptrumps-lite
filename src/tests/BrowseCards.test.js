import BrowseCards from "../components/BrowseCards";
import { render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("BrowseCards component", () => {
  test("renders correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <BrowseCards />
      </BrowserRouter>
    );
    expect(asFragment).toMatchSnapshot();
  });

  test("shows stats of card from select menu when clicked", async () => {
    const { getByText } = render(
      <BrowserRouter>
        <BrowseCards />
      </BrowserRouter>
    );

    fireEvent.change(document.getElementById("selectCards"), {
      target: { value: "Cat" },
    });
    expect(getByText('"Tiddles"')).toBeTruthy();
  });

  test("shows stats of card from mobile dropdown menu when clicked", async () => {
    const { getByText } = render(
      <BrowserRouter>
        <BrowseCards />
      </BrowserRouter>
    );

    fireEvent.change(document.getElementById("selectCards-drop"), {
      target: { value: "Cat" },
    });
    expect(getByText('"Tiddles"')).toBeTruthy();
  });
});
