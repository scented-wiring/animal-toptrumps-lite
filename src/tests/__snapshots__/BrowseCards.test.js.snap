// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseCards component renders correctly 1`] = `[Function]`;
